interop-testing/src/main/java/io/grpc/testing/integration/StressTestClient.java

```diff
@@ -0,0 +1,574 @@
+          break;
+
+        case CLIENT_STREAMING:
+          tester.clientStreaming();
+          break;
+
+        case SERVER_STREAMING:
+          tester.serverStreaming();
+          break;
+
+        case PING_PONG:
+          tester.pingPong();
+          break;
+
+        case EMPTY_STREAM:
+          tester.emptyStream();
+          break;
+
+        case UNIMPLEMENTED_METHOD: {
+          tester.unimplementedMethod();
+          break;
+        }
+
+        case CANCEL_AFTER_BEGIN: {
+          tester.cancelAfterBegin();
+          break;
+        }
+
+        case CANCEL_AFTER_FIRST_RESPONSE: {
+          tester.cancelAfterFirstResponse();
+          break;
+        }
+
+        case TIMEOUT_ON_SLEEPING_SERVER: {
+          tester.timeoutOnSleepingServer();
+          break;
+        }
+
+        default:
+          throw new IllegalArgumentException("Unknown test case: " + testCase);
+      }
+    }
+
+    class Tester extends AbstractInteropTest {
+      @Override
+      protected ManagedChannel createChannel() {
+        return Worker.this.channel;
+      }
+    }
+
+    class WeightedTestCaseSelector {
+      // Randomly shuffled list that contains each testcase as often as
+      // the testcase's weight.
+      final List<TestCases> testCases = new ArrayList<TestCases>();
+      int idx;
+
+      WeightedTestCaseSelector(List<TestCaseWeightPair> testCaseWeightPairs) {
+        for (TestCaseWeightPair testCaseWeightPair : testCaseWeightPairs) {
+          for (int i = 0; i < testCaseWeightPair.weight; i++) {
+            testCases.add(testCaseWeightPair.testCase);
+          }
+        }
+
+        shuffle(testCases);
+      }
+
+      TestCases nextTestCase() {
+        return testCases.get(idx++ % testCases.size());
+      }
+    }
+  }
+
+  /**
+   * Service that exports the QPS metrics of the stress test.
+   */
+  private class MetricsServiceImpl implements MetricsServiceGrpc.MetricsService {
+
+    @Override
+    public void getAllGauges(Metrics.EmptyMessage request,
+        StreamObserver<Metrics.GaugeResponse> responseObserver) {
+      for (Metrics.GaugeResponse gauge : gauges.values()) {
+        responseObserver.onNext(gauge);
+      }
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getGauge(Metrics.GaugeRequest request,
+        StreamObserver<Metrics.GaugeResponse> responseObserver) {
+      String gaugeName = request.getName();
+      Metrics.GaugeResponse gauge = gauges.get(gaugeName);
+      if (gauge != null) {
+        responseObserver.onNext(gauge);
+        responseObserver.onCompleted();
+      } else {
+        responseObserver.onError(new Throwable("Gauge '" + gaugeName + "' does not exist."));
+      }
+    }
+  }
+
+  @VisibleForTesting
+  static class TestCaseWeightPair {
+    final TestCases testCase;
+    final int weight;
+
+    TestCaseWeightPair(TestCases testCase, int weight) {
+      this.testCase = Preconditions.checkNotNull(testCase);
```

Comment:
Question: some constructors in this PR call preconditions and some don't  Any reason for the discrepancy?